src/rSW_Site.c

```diff
@@ -243,10 +245,29 @@ static void onSet_SW_SWRCp(SEXP SW_SWRCp, LOG_INFO* LogInfo) {
 	p_SWRCp = REAL(SW_SWRCp);
 
 	for (i = 0; i < (v->n_layers); i++) {
+        if (!isMineral && i > 1) {
+            /* Fibric and sapric peat are completed.
+            Now: reset and restart for swrcp of the mineral component */
+            isMineral = swTRUE;
+            i = 0;
+        }
+
+        if (isMineral && !v->inputsProvideSWRCp) {
+            return;
+        }
+
 		for (k = 0; k < SWRC_PARAM_NMAX; k++) {
-			v->swrcp[i][k] = p_SWRCp[i + (v->n_layers) * k];
+            if (isMineral) {
+                v->soils.swrcpMineralSoil[i][k] = p_SWRCp[i + (v->n_layers) * k];
+            } else {
+                v->swrcpOM[i][k] = p_SWRCp[i + (v->n_layers) * k];

```

Comment:
onSet_SW_SWRCp() -- see longer comment for onGet_SW_SWRCp()
- copy `p_SWRCp` into `v->soils.swrcpMineralSoil`
- copy new "omSWRCp" into `v->swrcpOM`